### DISCUSSION TITLE
Attempting To Enable Software Serial Results In Black Screen

### DISCUSSION THREAD

**Discussion by TehProgrammer** [Jyers/Marlin#1346] [Q&A] [answered]
I've made the hardware modifications as specified [here](https://drive.google.com/file/d/15cUf10lMxW4NUHE9qVAlamXmULdrOJxz/view).

This modification was confirmed to NOT break anything, as I can upload new, prebuilt firmware to it without issue...

I've also tried the software modifications on multiple Jyers source codes and configurations (and more recently, direct from main and bugfix branch Marlin).  I'm able to get the printer to flash the firmware UNTIL I change the serial settings.  As soon as that changes... Black screen and no flashing (now just not compiling).  This can be reversed by simply flashing a previously working FW.  

Is there something with Jyers FW that conflicts with enabling SW Serial on PA3?

Files I changed can be found below (only one example, just tried direct from Marlin with no success):
[pins_CREALITY_V4.h.txt](https://github.com/Jyers/Marlin/files/7069616/pins_CREALITY_V4.h.txt)
[platformio.ini.txt](https://github.com/Jyers/Marlin/files/7069617/platformio.ini.txt)
[Configuration.h.txt](https://github.com/Jyers/Marlin/files/7069618/Configuration.h.txt)
[Configuration_adv.h.txt](https://github.com/Jyers/Marlin/files/7069619/Configuration_adv.h.txt)

**NEW ISSUE:**
Cannot compile due to the software serial library I'm using not having an expected function:
error: 'setInterruptPriority' is not a member of 'SoftwareSerial'  (using softwareserialm)

**Comment by pram0** [comment, marked as answer]
i was also having the same problem, i solved the issue by using PA13 or PA14 instead of PA3. this will work for if you just want to enable UART on 1 or 2 drivers. I want all driver with UART enabled. IF i use PA3,PA4,PA7 i can compile the firmware but when i go to flash i just gets stuck on blank screen.

**Reply by pram0** [reply]
finnally managed to get all drivers with UART enabled, i was using the latest marlin bugfix release and for whatever reason it would not work in uart for pin pa3,pa4,pa7 and i would just get the black screen when flashing using any of those pins.downloaded marlin 2.0.8 and issue was solved, now i have uart on all drivers im using pa13,pa14,pa3, pa7.

**Reply by EntasDisk** [reply]
Hello, do you have working firmware? BLTouch
No matter what I do, I always get an error. I don't have any in my code info.
